schutzbot/mockbuild.sh

```diff
@@ -51,7 +54,7 @@ if [[ $ID == rhel || $ID == centos ]] && ! rpm -q epel-release; then
 fi
 
 # Register RHEL if we are provided with a registration script.
-if [[ -n "${RHN_REGISTRATION_SCRIPT:-}" ]] && ! sudo subscription-manager status; then
+if [[ $ID == "rhel" && $VERSION_ID == "8.3" && -n "${RHN_REGISTRATION_SCRIPT:-}" ]] && ! sudo subscription-manager status; then
```

Comment:
Why change the condition? The current implementation will need maintenance as soon as we switch to a newer version. What's wrong with the old if condition ?

Reply:
In Jenkins the script was supplied only to stages which ran on rhel-8 and here in Gitlab CI it's present all the time, so this prevents it from running on other runners... But yeah this is not a permanent solution, I'll think it over.

Reply:
The runners config.json contains `subscriptionNeeded` which is used when installing gitlab-runner RPM. Maybe we can export this so that it is available to the rest of the scripts ?

Reply:
I don't know how. It just checks for `subscriptionNeeded` when preparing the custom runner and the only runner that has it now is rhel-8-x86_64 in openstack... I think the easiest way would be to specify such a variable in `.gitlab-ci.yml` and check for that, similar to how we set `INTERNAL_NETWORK` variable...

Reply:
I just changed it to rely on setting a variable `REGISTER=true`  in `.gitlab-ci.yml` .. It's basically the same as setting the environment for stages in Jenkinsfile.